pkg/conversion/converter.go

```diff
@@ -296,29 +331,17 @@ func verifyConversionFunctionSignature(ft reflect.Type) error {
 //                 return nil
 //          })
 func (c *Converter) RegisterConversionFunc(conversionFunc interface{}) error {
-	fv := reflect.ValueOf(conversionFunc)
-	ft := fv.Type()
-	if err := verifyConversionFunctionSignature(ft); err != nil {
-		return err
-	}
-	c.conversionFuncs[typePair{ft.In(0).Elem(), ft.In(1).Elem()}] = fv
-	return nil
+	return c.conversionFuncs.Register(conversionFunc)
 }
 
 // Similar to RegisterConversionFunc, but registers conversion function that were
 // automatically generated.
 func (c *Converter) RegisterGeneratedConversionFunc(conversionFunc interface{}) error {
-	fv := reflect.ValueOf(conversionFunc)
-	ft := fv.Type()
-	if err := verifyConversionFunctionSignature(ft); err != nil {
-		return err
-	}
-	c.generatedConversionFuncs[typePair{ft.In(0).Elem(), ft.In(1).Elem()}] = fv
-	return nil
+	return c.generatedConversionFuncs.Register(conversionFunc)
 }
 
 func (c *Converter) HasConversionFunc(inType, outType reflect.Type) bool {
-	_, found := c.conversionFuncs[typePair{inType, outType}]
+	_, found := c.conversionFuncs.fns[typePair{inType, outType}]
```

Comment:
Yeah, would be cleaner



On Dec 1, 2015, at 7:18 PM, Daniel Smith <notifications@github.com> wrote:

In pkg/conversion/converter.go
<https://github.com/kubernetes/kubernetes/pull/17922#discussion_r46361017>:

>  }
>
>  func (c *Converter) HasConversionFunc(inType, outType reflect.Type) bool {
> -	_, found := c.conversionFuncs[typePair{inType, outType}]
> +	_, found := c.conversionFuncs.fns[typePair{inType, outType}]

Optional: Consider making this a method of ConversionFuncs?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17922/files#r46361017>.